File: library/src/main/java/me/proxer/library/api/DefaultLoginTokenManager.java
package me.proxer.library.api;

import javax.annotation.Nullable;

/**
 * @author Ruben Gees
 */
final class DefaultLoginTokenManager implements LoginTokenManager {

    private String currentToken;

    @Override
    @Nullable
    public String provide() {
        return currentToken;
    }

    @Override
    public void persist(@Nullable final String loginToken) {
        currentToken = loginToken;
    }
}

File: library/src/main/kotlin/me/proxer/library/api/DefaultLoginTokenManager.kt
package me.proxer.library.api

/**
 * @author Ruben Gees
 */
internal class DefaultLoginTokenManager : LoginTokenManager {

    @Volatile
    private var currentToken: String? = null

    override fun provide(): String? {
        return currentToken
    }

    override fun persist(loginToken: String?) {
        currentToken = loginToken
    }
}
